token-locker/src/lib.rs

```diff
@@ -274,8 +324,10 @@ impl Contract {
         #[serializer(borsh)] proof: Proof,
     ) -> Promise {
         assert!(verification_success, "Failed to verify the proof");
-        let required_deposit = self.record_proof(&proof);
 
+        // The deposit required to cover the cost
+        // of additional storage of used proof
+        let required_deposit = self.record_proof(&proof);
```

Comment:
What if we get an error during ft_transfer_call? In that case, we will record that proof is already used, but we don't unlock tokens.